dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI/State/TurnStateEntry.cs

```diff
@@ -6,10 +6,11 @@
 namespace Microsoft.Teams.AI.State
 {
     /// <summary>
-    /// The turn state entry.
+    /// Accessor class for managing an individual state scope.
     /// </summary>
-    /// <typeparam name="TValue">The type of value.</typeparam>
-    public class TurnStateEntry<TValue> : IReadOnlyEntry<TValue> where TValue : class
+    /// <typeparam name="TValue">Strongly typed value of the state scope.</typeparam>
+    public class TurnStateEntry<TValue> : IReadOnlyEntry<TValue>
```

Comment:
Throughout the application the only generic type used is `Record`. The `_value` type can just be `Record`. Plus this would make C# more in parity with JS (as it also isn't a generic class)